### PR TITLE
Add survey info in the GTM data_layer

### PR DESCRIPTION
**Pull request by petechd**
### What is the context of this PR?
Adds new keys to business schemas `dataLayer`. Check [trello card](https://trello.com/c/iN8EKg5k).

### How to review 
Extra integration tests added for different scenarios.

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

**Comment by petechd** [comment]
[Survey info in the GTM data_layer (2 SP)](https://trello.com/c/iN8EKg5k/5246-survey-info-in-the-gtm-datalayer-2-sp)

File: app/survey_config/business_config.py [hunk 1]
```diff
@@ -17,6 +17,14 @@ class BusinessSurveyConfig(
     footer_legal_links: Iterable[Mapping] = field(default_factory=list)
 
     def __post_init__(self):
+        if self.schema:
+            self.data_layer: Iterable[Mapping] = [
```
Comment (MebinAbraham) [inline app/survey_config/business_config.py:21]: Anything we want to do with a property that is inherited should be after the `post_init` call, i.e post init might override or do something with it. Prior only if we want to preprocess before calling post init, such as the base url.

**Review by MebinAbraham — COMMENTED** [review]
Functionally as expected, just need confirmation it is as expected in GA.

File: app/survey_config/business_config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (petechd) [reply app/survey_config/business_config.py]: Changed, yeah it is literally just a list of dicts (one dict atm).

Comment (petechd) [reply app/survey_config/business_config.py]: Moved after `post_init`, not always 100% how it works, thanks.

File: app/helpers/template_helpers.py [hunk 2]
```diff
@@ -178,13 +184,21 @@ def get_survey_config(
...
 
 
 def render_template(template: str, **kwargs: Union[str, Mapping]) -> str:
     language = get_locale().language
+    session_store = get_session_store()
+    if session_store:
+        # pylint: disable=assigning-non-slot
+        schema = load_schema_from_session_data(session_store.session_data)
+    else:
+        schema = None
```
[the remark this answers is not in the document]
Comment (petechd) [reply app/helpers/template_helpers.py]: Modified it so gets extra `session_data` check but also two walruses.

File: app/helpers/template_helpers.py [hunk 3]
```diff
@@ -178,13 +184,21 @@ def get_survey_config(
         theme=theme,
         language=language,
         base_url=base_url,
+        schema=schema,
     )
 
 
 def render_template(template: str, **kwargs: Union[str, Mapping]) -> str:
     language = get_locale().language
+    session_store = get_session_store()
```
[the remark this answers is not in the document]
Comment (petechd) [reply app/helpers/template_helpers.py]: Ok, using one from line 193 in 217 now, thanks.

File: app/helpers/template_helpers.py [hunk 4]
```diff
@@ -178,13 +184,21 @@ def get_survey_config(
...
 
 
 def render_template(template: str, **kwargs: Union[str, Mapping]) -> str:
     language = get_locale().language
+    if session_store := get_session_store():
+        if session_data := session_store.session_data:
+            # pylint: disable=assigning-non-slot
+            schema = load_schema_from_session_data(session_data)
+    else:
+        schema = None
```
Comment (MebinAbraham) [inline app/helpers/template_helpers.py:198]: - We need session store for both schema and expiry time, therefore, we should do them in the same place. 
- Also, at the moment, schema is undefined if session data doesn't exist, and would result in an error.
- The pylint ignore is redundant.

We can do something like this:
```py
    schema, session_expires_at = None, None
    if session_store := get_session_store():
        if session_data := session_store.session_data:
            schema = load_schema_from_session_data(session_data)

        if session_expiry := session_store.expiration_time:
            session_expires_at = session_expiry.isoformat()
```

The session expiry assignment on line 218 can then be removed.

File: tests/integration/test_application_variables.py [hunk 5]
```diff
@@ -16,18 +16,32 @@ def tearDown(self):
         settings.EQ_GOOGLE_TAG_MANAGER_AUTH = None
 
     def test_google_analytics_code_and_credentials_are_present(self):
-        self.launchSurvey("test_textfield")
+        self.launchSurvey("test_feedback")
```
Comment (MebinAbraham) [inline tests/integration/test_application_variables.py:19]: Any reason for the schema change?

Comment (petechd) [reply tests/integration/test_application_variables.py:19]: Yeah, the feedback schema has a form type, the other one doesn't or the other way round.

File: app/helpers/template_helpers.py (hunk 4 above)
Comment (petechd) [reply app/helpers/template_helpers.py]: Ok, fixed, thanks.

Comment (MebinAbraham) [reply app/helpers/template_helpers.py]: > The session expiry assignment on line 218 can then be removed.

Still needs doing.

File: app/helpers/template_helpers.py [hunk 6]
```diff
@@ -178,13 +184,22 @@ def get_survey_config(
...
 def render_template(template: str, **kwargs: Union[str, Mapping]) -> str:
     language = get_locale().language
+    schema, session_expires_at = None, None
+    if session_store := get_session_store():
+        if session_data := session_store.session_data:
+            schema = load_schema_from_session_data(session_data)
+
+        if session_expiry := session_store.expiration_time:
+            session_expires_at = session_expiry.isoformat()
     survey_config = get_survey_config(
```
Comment (MebinAbraham) [inline app/helpers/template_helpers.py:200]: New line would help readability.

File: app/survey_config/business_config.py [hunk 7]
```diff
@@ -20,6 +20,15 @@ def __post_init__(self):
...
         super().__post_init__()
 
+        if self.schema:
+            self.data_layer: list[dict] = [
+                {
+                    key: self.schema.json[key]
+                    for key in ["form_type", "survey_id", "title"]
+                    if key in self.schema.json
+                }
+            ]
```
Comment (MebinAbraham) [inline app/survey_config/business_config.py:30]: Could use walrus operator:
```suggestion
            self.data_layer: list[dict] = [
                {
                    key: value
                    for key in ["form_type", "survey_id", "title"]
                    if (value := self.schema.json.get(key))
                }
            ]
```

File: app/helpers/template_helpers.py (hunk 6 above)
Comment (petechd) [reply app/helpers/template_helpers.py]: Ok, added new line, thanks!

File: app/helpers/template_helpers.py (hunk 4 above)
Comment (petechd) [reply app/helpers/template_helpers.py]: Misunderstood this one, ok removed this assignment now.

File: tests/functional/spec/exit.spec.js [hunk 8]
```diff
@@ -1,33 +0,0 @@
-import CensusThankYouPage from "../base_pages/census-thank-you.page.js";
```
Comment (MebinAbraham) [inline tests/functional/spec/exit.spec.js:1]: How come this file was removed?

**Review by MebinAbraham — COMMENTED** [review]
Looks good, just need confirmation it is working in GTM.

File: tests/functional/spec/exit.spec.js (hunk 8 above)
Comment (petechd) [reply tests/functional/spec/exit.spec.js]: I was testing something and pushed it up with other changes, reverted it now.

**Comment by MebinAbraham** [comment]
Once confirmation has been given, this is working after integrating with GTM, this can be merged.

**Review by berroar — APPROVED** [review]

**Pull request merged**